ui/src/app/services/sign-in.service.ts

```diff
@@ -69,68 +50,32 @@ export class SignInService {
             client_id: this.clientId,
             hosted_domain: config.gsuiteDomain,
             scope: 'https://www.googleapis.com/auth/plus.login openid profile'
+        }).then(() => {
+          this.subscribeToAuth2User();
         });
         resolve(gapi.auth2);
       });
     });
   }
 
-  private makeUserSubject(): Observable<SignInDetails> {
-    const initialDetailsString = window.sessionStorage.getItem(SIGNED_IN_USER);
-    const initialDetails: SignInDetails = initialDetailsString ?
-      JSON.parse(initialDetailsString) : {isSignedIn: false};
-    return new BehaviorSubject(initialDetails);
-  }
-
   private subscribeToAuth2User(): void {
-    this.auth2.then(auth2 => {
-      gapi.auth2.getAuthInstance().currentUser.listen((e: any) => {
-        const currentUser = gapi.auth2.getAuthInstance().currentUser.get();
-        const details = this.extractSignInDetails(currentUser);
-        // Without this, Angular views won't "notice" the externally-triggered
-        // event, though the Angular models will update... so the change
-        // won't propagate to UI automatically. Calling `zone.run`
-        // ensures Angular notices as soon as the change occurs.
-        const user: BehaviorSubject<SignInDetails> = this.user as BehaviorSubject<SignInDetails>;
-        this.zone.run(() => user.next(details));
+    if (gapi.auth2.getAuthInstance().isSignedIn.get()) {
+      this.zone.run(() => {
+        this.isSignedIn.next(true);
+      });
+    }
+    gapi.auth2.getAuthInstance().isSignedIn.listen((isSignedIn: boolean) => {
+      this.zone.run(() => {
```

Comment:
Is the purpose of running these in the zone to force change detection?

Reply:
Yes. I think. It broke without the zone, and I was wondering why it broke, and adding the zone fixed it. I had taken out zone.run in earlier attempts, so I tried adding it back in and it all worked again. Not entirely sure why, but I think I still only have a cursory knowledge of zones.